odoo/addons/base/models/ir_http.py

```diff
@@ -238,168 +240,160 @@ def _clear_routing_map(cls):
         if hasattr(cls, '_routing_map'):
             del cls._routing_map
 
-    @classmethod
-    def content_disposition(cls, filename):
-        return content_disposition(filename)
+    #------------------------------------------------------
+    # Binary server
+    #------------------------------------------------------
 
     @classmethod
     def _xmlid_to_obj(cls, env, xmlid):
         return env.ref(xmlid, False)
 
-    @classmethod
-    def _check_access_mode(cls, env, id, access_mode, model, access_token=None, related_id=None):
-        """
-        Implemented by each module to define an additional way to check access.
-
-        :param env: the env of binary_content
-        :param id: id of the record from which to fetch the binary
-        :param access_mode: typically a string that describes the behaviour of the custom check
-        :param model: the model of the object for which binary_content was called
-        :param related_id: optional id to check security.
-        :return: True if the test passes, else False.
-        """
-        return False
-
-
-    @classmethod
-    def binary_content(cls, xmlid=None, model='ir.attachment', id=None, field='datas',
-                       unique=False, filename=None, filename_field='datas_fname', download=False,
-                       mimetype=None, default_mimetype='application/octet-stream',
-                       access_token=None, related_id=None, access_mode=None, env=None):
-        """ Get file, attachment or downloadable content
-
-        If the ``xmlid`` and ``id`` parameter is omitted, fetches the default value for the
-        binary field (via ``default_get``), otherwise fetches the field for
-        that precise record.
-
-        :param str xmlid: xmlid of the record
-        :param str model: name of the model to fetch the binary from
-        :param int id: id of the record from which to fetch the binary
-        :param str field: binary field
-        :param bool unique: add a max-age for the cache control
-        :param str filename: choose a filename
-        :param str filename_field: if not create an filename with model-id-field
-        :param bool download: apply headers to download the file
-        :param str mimetype: mintype of the field (for headers)
-        :param related_id: the id of another record used for custom_check
-        :param  access_mode: if truthy, will call custom_check to fetch the object that contains the binary.
-        :param str default_mimetype: default mintype if no mintype found
-        :param str access_token: optional token for unauthenticated access
-                                 only available  for ir.attachment
-        :param Environment env: by default use request.env
-        :returns: (status, headers, content)
-        """
-        env = env or request.env
+    def _binary_security_check(self, xmlid=None, model=None, id=None, field='datas', access_token=None):
```

Comment:
This method returns a record, so its name should show it. How about `_get_record_checked()` or something similar?

Reply:
or `_binary_record_checked()` ?